src/components/TimeseriesChartExport/TimeseriesChartExport.tsx

```diff
@@ -0,0 +1,415 @@
+  ) => {
+    const [start, end] = form.getFieldValue('range');
+    const seriesNumber = timeseries.length;
+    const granularityString = event.currentTarget.value;
+    const granularityValue = getGranularityInSeconds(granularityString);
+
+    setLimitHit(
+      isGreaterThenLimit(limit, start, end, granularityValue, seriesNumber)
+    );
+  };
+
+  const checkLimitOnRangeChange = (range: RangePickerValue) => {
+    const granularityString = form.getFieldValue('granularity');
+    const seriesNumber = timeseries.length;
+    let start = 0;
+    let end = 0;
+    const granularityValue = getGranularityInSeconds(granularityString);
+
+    if (range[0] && range[1]) {
+      start = +range[0];
+      end = +range[1];
+    }
+
+    setLimitHit(
+      isGreaterThenLimit(limit, start, end, granularityValue, seriesNumber)
+    );
+  };
+
+  const onSubmit = async (event: React.SyntheticEvent) => {
+    event.preventDefault();
+
+    const limitPerSerie = Math.floor(limit / series.length);
+
+    form.validateFields(async (_, values) => {
+      const {
+        range,
+        granularity: granularityVal,
+        delimiter,
+        readableDate,
+      }: ChartExportFormFields = values;
+      const aggregate = 'average';
+      const body: DatapointsMultiQuery = {
+        items: series.map(({ id }) => ({
+          id,
+          aggregates: [aggregate],
+        })),
+        start: +range[0],
+        end: +range[1],
+        limit: limitPerSerie,
+        granularity: granularityVal,
+      };
+
+      setLoading(true);
+
+      try {
+        const csvString = fetchCSV
+          ? await fetchCSV(body, {
+              aggregate,
+              delimiter,
+              readableDate,
+              granularity: granularityVal,
+            })
+          : await fetchCSVCall(body, {
+              aggregate,
+              delimiter,
+              readableDate,
+              granularity: granularityVal,
+            });
+
+        setLoading(false);
+
+        if (onSuccess) {
+          onSuccess();
+        }
+
+        downloadCSV(csvString);
+      } catch (e) {
+        setLoading(false);
+
+        if (onError) {
+          onError(e);
+        }
+      }
+    });
+  };
+
+  useEffect(() => {
+    if (cellLimit > CELL_LIMIT) {
+      setLimit(CELL_LIMIT);
+    }
+  }, [cellLimit]);
+
+  useEffect(() => {
+    const { range, granularity: granularityString } = form.getFieldsValue();
+    const start = range[0] ? +range[0] : 0;
+    const end = range[1] ? +range[1] : 0;
+    const granularityValue = getGranularityInSeconds(granularityString);
+    const seriesNumber = timeseries.length;
+
+    setLimitHit(
+      isGreaterThenLimit(limit, start, end, granularityValue, seriesNumber)
+    );
+  }, []);
+
+  useEffect(() => {
+    if (!visible || !timeseries.length) {
+      return;
+    }
+
+    const request = timeseries.map(tid =>
+      typeof tid === 'number' ? { id: tid } : { externalId: tid }
```

Comment:
I would go for `timeseriesIds` for now, this logic seems not very safe to me.
And this way it will also be aligned with other components

Reply:
done